libmamba/src/core/package_handling.cpp

```diff
@@ -57,7 +58,104 @@ namespace mamba
         const fs::u8path& m_file;
     };
 
-    static int copy_data(archive* ar, archive* aw)
+    class libarchive_read_raii
+    {
+    public:
+        libarchive_read_raii()
+            : libarchive_read_raii(archive_read_new()){};
+        static libarchive_read_raii read_disk()
+        {
+            return libarchive_read_raii(archive_read_disk_new());
+        }
+
+        ~libarchive_read_raii()
+        {
+            archive_read_free(m_archive);
+        }
+
+        operator archive*()
```

Comment:
make it `explicit` otherwise conversion to int are possible

also need a const version

Personally I would avoid operators altogether and just provide an accessor, either `operator*` or `archive* raw()` (or similar name) or both.
Same remark for the other types.

Reply:
Using an explict accessor would modify a lot of functions and that's not the goal of this PR. Let's do that later if we want to.

Reply:
actually the `explicit` doesn't work for `archive*` operator because it can't be converted to `struct archive*` ...

Reply:
ok for separate pr though I dont see how changing it here would change previously written functions?

> actually the explicit doesn't work for archive* operator because it can't be converted to struct archive*

I didnt understand what you mean here?